sans/direct-beam.ipynb

```diff
@@ -0,0 +1,529 @@
+    "           mantid_args={'FirstColumnValue':'Wavelength'})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "sample_trans = load_larmor(sample_transmission_run_number)\n",
+    "sample = load_larmor(sample_run_number)\n",
+    "background_trans = load_larmor(background_transmission_run_number)\n",
+    "background = load_larmor(background_run_number)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizations and mask setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "masks = {}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Bad straws"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from scipp.plot import plot\n",
+    "from loki import LoKI\n",
+    "loki = LoKI()\n",
+    "spectrum_counts = sc.sum(sample, 'tof') # sum is optional, could also keep TOF\n",
+    "pixel_counts = loki.to_logical_dims(spectrum_counts)\n",
+    "straw_counts = sc.sum(pixel_counts, 'pixel')\n",
+    "plot(straw_counts, log=True, vmin=1, vmax=7)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bad_straws = sc.Variable(dims=['spectrum'], shape=sample.coords['spectrum'].shape, dtype=sc.dtype.bool)\n",
+    "mask = sc.Variable(value=True)\n",
+    "bad_straws[loki.straw(tube=0, straw=6)] |= mask\n",
+    "bad_straws[loki.straw(tube=1, straw=2)] |= mask\n",
+    "bad_straws[loki.straw(tube=5, straw=1)] |= mask\n",
+    "for straw in 1,2,3,4,5:\n",
+    "    bad_straws[loki.straw(tube=12, straw=straw)] |= mask\n",
+    "masks['bad-straws'] = bad_straws"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Bad pixels"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plot(pixel_counts, vmax=100, axes=['straw', 'tube', 'pixel'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pos = sc.neutron.position(sample)\n",
+    "x = sc.geometry.x(pos)\n",
+    "y = sc.geometry.y(pos)\n",
+    "counts = spectrum_counts.data\n",
+    "counts.variances = None # TODO support comparison with variances\n",
+    "masks['electronics-error'] = sc.less(sc.abs(x), 0.2 * sc.units.m) \\\n",
+    "                             & sc.less(sc.abs(y), 0.03 * sc.units.m) \\\n",
+    "                             & sc.equal(counts, 0.0 * sc.units.counts)\n",
+    "print(f\"Masking {sc.sum(masks['electronics-error'], 'spectrum').value} bad pixels due to electronics error.\")"
```

Comment:
Very nice logical form of combining and applying spacial and value based mask critera.

I believe this produces no masked elements in it's current form with the input data from the notebook. Why do we need it?

Reply:
It does with the updated files (not on github).